compositor_render/src/transformations/transition.rs

```diff
@@ -0,0 +1,99 @@
+use std::time::Duration;
+
+use compositor_common::{
+    scene::{
+        transition::{Interpolation, TransitionSpec, TransitionType},
+        NodeId, Resolution,
+    },
+    util::InterpolationState,
+};
+use log::error;
+
+use crate::{
+    error::CreateNodeError,
+    renderer::{texture::NodeTexture, RenderCtx},
+};
+
+use super::builtin::{BuiltinNode, BuiltinState, BuiltinTransition};
+
+pub struct TransitionNode {
+    builtin_node: BuiltinNode,
```

Comment:
> We wanted to add transitions to shader nodes as well.

Eventually, yes

> Wouldn't this be problematic?

Why? Adding support for shader would just be wrapping this node into enum and implementing the other variant.

> but since at other places in this PR whole architecture is designed to use shader nodes as well, I don't get why this isn't the case here.

- This PR has to have some scope, it's already grew more than I would prefer
- Shader support is not that important, it's good to have but we don't need it now
- Current implementation is written in a way that adding shader support does not require any refactor in code that is added in this PR. To add this all that you need is to
  - replace BuiltInNode with some enum wrapper
  - implement generic interpolation for shader params
  - add logic in shader node for updating params

Reply:
Ok